src/sys/petsclib.cxx

```diff
@@ -25,8 +27,25 @@ PetscLib::PetscLib() {
       PetscInitialize(pargc,pargv,PETSC_NULL,help);
       PetscLogEventRegister("Total BOUT++",0,&USER_EVENT);
       PetscLogEventBegin(USER_EVENT,0,0,0,0);
+
+    // Load global PETSc options from the [petsc] section of the input
+    setPetscOptions(Options::root()["petsc"], "");
+  }
+
+  if (opt != nullptr and opt->isSection()) {
+    // Use options specific to this PetscLib
+    // Pass options to PETSc's global options database, with a unique prefix, that will be
+    // passed to a KSP later.
+    // (PetscOptions type exists for non-global options, but apparently is only for user
+    // options, and cannot be passed to KSP, etc. Non-global options can be passed by
+    // defining a custom prefix for the options string, and then passing that to the KSP.)
+
+    options_prefix = "boutpetsclib" + std::to_string(unique_id) + "_";
```

Comment:
Could we somehow pass a `name` in `opt` down here, and only fallback to this if we don't have one? 
Could we even fallback to the section name? Then we could avoid the unique ID altogether perhaps?

`Options::str()` actually just returns the full name, including path from root, so this should be globally unique. This is probably more ergonomic, as then you should just be able to do something like `aparsolver:petsc:ksp_type=gmres` on the command line.

I could be missing something here!

Reply:
I think using `opt->str()` is the best solution. If more than one solver gets its options from the same section, we might reset some options, but because they come from the same section, we'd only reset them to the same value so that should be fine.

While I was tidying up, I also moved getting the `"petsc"` subsection into `PetscLib` instead of getting it at the call-sites, which simplifies the rest of the code, and enforces a standardisation of the subsection naming.